### PR TITLE
m.presence can be null in THREAD_MEMBERS_UPDATE

### PR DESCRIPTION
**Pull request by ZixeSea**
In the THREAD_MEMBERS_UPDATE event, the user is updated what also includes a presence update (m.presence). The m.presence object can be null resulting in a "Cannot read property 'id' of null" error.

### CODE REVIEW

**Review by bsian03 — DISMISSED** [review]

**Review by RaZeSloth — DISMISSED** [review]

File: lib/structures/Interaction.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ class Interaction extends Base {
             }
         }
 
-        this._client.emit("warn", new Error(`Unknown interaction type: ${data.type}\n${JSON.stringify(data)}`));
+        client.emit("warn", new Error(`Unknown interaction type: ${data.type}\n${JSON.stringify(data)}`));
```
Comment (HeadTriXz) [inline lib/structures/Interaction.js]: Unrelated changes should be in a separate pull request.

File: lib/gateway/Shard.js [hunk 2]
```diff
@@ -2243,7 +2243,9 @@ class Shard extends EventEmitter {
                         m.member.id = m.member.user.id;
                         const guild = this.client.guilds.get(packet.d.guild_id);
                         if(guild) {
-                            guild.members.update(m.presence, guild);
+                            if(m.presence) {
+                                guild.members.update(m.presence, guild);
+                            }
```
Comment (HeadTriXz) [inline lib/gateway/Shard.js:2248]: Looks good to me.

**Review by HeadTriXz — DISMISSED** [review]

File: lib/structures/Interaction.js (hunk 1 above)
Comment (DonovanDMC) [reply lib/structures/Interaction.js]: I've got this in #1309

**Review by HeadTriXz — APPROVED** [review]

File: lib/gateway/Shard.js [hunk 3]
```diff
@@ -2243,7 +2243,9 @@ class Shard extends EventEmitter {
...
                             guild.members.update(m.member);
```
Comment (GweeKwee) [inline lib/gateway/Shard.js:2249]: This will throw a "Missing client in constructor" in user structure too
`guild.members.update(m.member);`

According to other places where the member is update, this should be:
`guild.members.update(m.member, guild);`

Comment (bsian03) [reply lib/gateway/Shard.js:2249]: Unrelated to this PR

Comment (GweeKwee) [reply lib/gateway/Shard.js:2249]: Do I need to open a pr for this? ><

Comment (ZixeSea) [reply lib/gateway/Shard.js:2249]: Making a new PR should be fine

**Review by bsian03 — APPROVED** [review]
